src/JustSaying/Fluent/QueuePublicationBuilder`1.cs

```diff
@@ -67,9 +72,56 @@ public QueuePublicationBuilder<T> WithWriteConfiguration(Action<SqsWriteConfigur
         }
 
         /// <inheritdoc />
-        void IPublicationBuilder<T>.Configure(JustSayingFluently bus)
+        void IPublicationBuilder<T>.Configure(
+            JustSayingBus bus,
+            IAwsClientFactoryProxy proxy,
+            ILoggerFactory loggerFactory)
         {
-            bus.WithSqsMessagePublisher<T>(ConfigureWrites);
+            var logger = loggerFactory.CreateLogger<QueuePublicationBuilder<T>>();
+
+            logger.LogInformation("Adding SQS publisher for message type '{MessageType}'.",
+                typeof(T));
+
+            var config = bus.Config;
+
+            var writeConfiguration = new SqsWriteConfiguration();
+            ConfigureWrites?.Invoke(writeConfiguration);
+            writeConfiguration.ApplyQueueNamingConvention<T>(config.QueueNamingConvention);
+
+            foreach (var region in config.Regions)
+            {
+                var regionEndpoint = RegionEndpoint.GetBySystemName(region);
+                var sqsClient = proxy.GetAwsClientFactory().GetSqsClient(regionEndpoint);
+
+                var eventPublisher = new SqsPublisher(
+                    regionEndpoint,
+                    writeConfiguration.QueueName,
+                    sqsClient,
+                    writeConfiguration.RetryCountBeforeSendingToErrorQueue,
+                    bus.SerializationRegister,
+                    loggerFactory)
+                {
+                    MessageResponseLogger = config.MessageResponseLogger
+                };
+
+                async Task StartupTask()
+                {
+                    logger.LogInformation("Beginning startup task to ensure queue {QueueName} exists", eventPublisher.QueueName);
```

Comment:
Maybe we should log it being done too?

Reply:
I think we can remove this log message, as each of the `ExistsAsync` and `CreateAsync` calls log their operations with timings:

`Checking if queue '{QueueName}' exists` followed by 
`Creating error queue {QueueName}`

Reply:
We're missing one on the actual Create though - theres's only one on the error queue creation so will add one there too.

Reply:
Done!